api/receiver/azure.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2018 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package receiver
+
+import (
+	"archive/zip"
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"regexp"
+)
+
+type azureArtifact struct {
+	name string
+	data []byte
+	z    *zip.Reader
+}
+
+func newAzureArtifact(artifactName string, zip io.Reader) (azureArtifact, error) {
+	data, err := ioutil.ReadAll(zip)
+	return azureArtifact{
+		name: artifactName,
+		data: data,
+	}, err
+}
+
+// extractReports extracts report files from the given zip.
+func (a azureArtifact) getReportFiles() ([]*zip.File, error) {
+	reportPath, err := regexp.Compile(fmt.Sprintf(`%s/wpt_report.*\.json$`, a.name))
```

Comment:
In general, I'd like to avoid constructing a regex from user input as it is a vector for DoS.

In this case, I think we only need to match the last component, so we can totally construct a global static regex.